those_names.py

```diff
@@ -0,0 +1,21 @@
+import string
+from itertools import product
+
+
+def full_names(first, last, min_length=0):
+    """
+    :param first: list of first names
+    :param last: list of last names
+    :param min_length: (optional) minimum length of name combination
+    :return: return all unique combinations of firstnames and lastnames
+    """
+    return [f[0].upper() + f[1::] + ' ' + l[0].upper() + l[1::] for f, l in list(product(first, last)) if (len(f) + len(l) >= min_length -1)]
```

Comment:
Why f and l? Avoid using single letter names for your variables, it makes your code harder for reading.